hazelcast/src/main/java/com/hazelcast/internal/dynamicclassloading/DynamicClassloadingService.java

```
@@ -0,0 +1,101 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.dynamicclassloading;
+
+import com.hazelcast.config.DynamicClassloadingConfig;
+import com.hazelcast.core.Member;
+import com.hazelcast.internal.dynamicclassloading.impl.ClassData;
+import com.hazelcast.internal.dynamicclassloading.impl.ClassLocator;
+import com.hazelcast.internal.dynamicclassloading.impl.ClassSource;
+import com.hazelcast.internal.dynamicclassloading.impl.filter.Filter;
+import com.hazelcast.internal.dynamicclassloading.impl.ClassDataProvider;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.ManagedService;
+import com.hazelcast.spi.NodeEngine;
+
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.hazelcast.internal.dynamicclassloading.impl.filter.ClassNameFilterParser.parseClassNameFilters;
+import static com.hazelcast.internal.dynamicclassloading.impl.filter.MemberProviderFilterParser.parseMemberFilter;
+
+public final class DynamicClassloadingService implements ManagedService {
+    public static final String SERVICE_NAME = "cl-service";
+
+    private boolean enabled;
+
+    private ClassDataProvider provider;
+    private ClassLocator locator;
+
+    @Override
+    public void init(NodeEngine nodeEngine, Properties properties) {
+        DynamicClassloadingConfig config = nodeEngine.getConfig().getDynamicClassloadingConfig();
+        if (!config.isEnabled()) {
+            //TODO: Fail-fast when some attributes are configured but the service is disabled
```

Not sure about this comment. Why fail fast? What if one disables the dynamic classloading intentionally? to test, to debug etc.